examples/multigrid-preconditioned-solver-customized/multigrid-preconditioned-solver-customized.cpp

```diff
@@ -0,0 +1,210 @@
+                                                 true);
+             }},
+            {"dpcpp",
+             [] {
+                 return gko::DpcppExecutor::create(
+                     0, gko::ReferenceExecutor::create());
+             }},
+            {"reference", [] { return gko::ReferenceExecutor::create(); }}};
+
+    // executor where Ginkgo will perform the computation
+    const auto exec = exec_map.at(executor_string)();  // throws if not valid
+
+    // Read data
+    auto A = share(gko::read<mtx>(std::ifstream("data/A.mtx"), exec));
+    // Create RHS as 1 and initial guess as 0
+    gko::size_type size = A->get_size()[0];
+    auto host_x = vec::create(exec->get_master(), gko::dim<2>(size, 1));
+    auto host_b = vec::create(exec->get_master(), gko::dim<2>(size, 1));
+    for (auto i = 0; i < size; i++) {
+        host_x->at(i, 0) = 0.;
+        host_b->at(i, 0) = 1.;
+    }
+    auto x = vec::create(exec);
+    auto b = vec::create(exec);
+    x->copy_from(host_x);
+    b->copy_from(host_b);
+
+    // Calculate initial residual by overwriting b
+    auto one = gko::initialize<vec>({1.0}, exec);
+    auto neg_one = gko::initialize<vec>({-1.0}, exec);
+    auto initres = gko::initialize<vec>({0.0}, exec);
+    A->apply(one, x, neg_one, b);
+    b->compute_norm2(initres);
+
+    // copy b again
+    b->copy_from(host_b);
+
+    // Prepare the stopping criteria
+    const gko::remove_complex<ValueType> tolerance = 1e-8;
+    auto iter_stop =
+        gko::share(gko::stop::Iteration::build().with_max_iters(100u).on(exec));
+    auto tol_stop = gko::share(gko::stop::ResidualNorm<ValueType>::build()
+                                   .with_baseline(gko::stop::mode::absolute)
+                                   .with_reduction_factor(tolerance)
+                                   .on(exec));
+    auto exact_tol_stop =
+        gko::share(gko::stop::ResidualNorm<ValueType>::build()
+                       .with_baseline(gko::stop::mode::rhs_norm)
+                       .with_reduction_factor(1e-14)
+                       .on(exec));
+
+    std::shared_ptr<const gko::log::Convergence<ValueType>> logger =
+        gko::log::Convergence<ValueType>::create();
+    iter_stop->add_logger(logger);
+    tol_stop->add_logger(logger);
+
+    // Now we customize some settings of the multigrid preconditioner.
+    // First we choose a smoother. Since the input matrix is spd, we use
+    // iterative refinement with two iterations and an Ic solver.
+    auto ic_gen = gko::share(
+        ic::build()
+            .with_factorization_factory(
+                gko::factorization::Ic<ValueType, int>::build().on(exec))
+            .on(exec));
+    auto smoother_gen = gko::share(
+        gko::solver::build_smoother(ic_gen, 2u, static_cast<ValueType>(0.9)));
+    // Use Pgm as the MultigridLevel factory (this is also the default).
+    auto mg_level_gen =
+        gko::share(pgm::build().with_deterministic(true).on(exec));
```

Comment:
Is it default mg_level? I do not see when it is set

Reply:
You're right, I will remove that comment